File: src/web/src/types/inventory-source.ts
// InventorySourceType - WHERE specimens are acquired from
export type InventorySourceType = 'Store' | 'Online' | 'Found' | 'Contact' | 'GemShow' | 'Other';

export interface InventorySourceDto {
  inventorySourceId: string;
  sourceType: InventorySourceType;
  name: string;
  location: string | null;
  phone: string | null;
  url: string | null;
  contactName: string | null;
  notes: string | null;
  isActive: boolean;
  dateCreated: string;
  dateUpdated: string;
  // Computed
  totalPurchases: number;
  lastPurchaseDate: string | null;
}

export interface InventorySourceListDto {
  inventorySourceId: string;
  sourceType: InventorySourceType;
  name: string;
  location: string | null;
  isActive: boolean;
  totalPurchases: number;
  lastPurchaseDate: string | null;
}

export interface InventorySourceSummaryDto {
  inventorySourceId: string;
  sourceType: InventorySourceType;
  name: string;
  location: string | null;
  phone: string | null;
  url: string | null;
  contactName: string | null;
}

export interface CreateInventorySourceRequest {
  sourceType: InventorySourceType;
  name: string;
  location?: string;
  phone?: string;
  url?: string;
  contactName?: string;
  notes?: string;
}

export interface UpdateInventorySourceRequest {
  sourceType: InventorySourceType;
  name: string;
  location?: string;
  phone?: string;
  url?: string;
  contactName?: string;
  notes?: string;
  isActive?: boolean;
}

// Filter options for inventory source list
export interface InventorySourceFilters {
  sourceType?: InventorySourceType;
  isActive?: boolean;
  search?: string;
  sortBy?: 'name' | 'sourceType' | 'location' | 'totalPurchases' | 'lastPurchaseDate' | 'dateCreated';
  sortOrder?: 'asc' | 'desc';
}

// Helper to get display name for source type
export const sourceTypeDisplayNames: Record<InventorySourceType, string> = {
  Store: 'Store',
  Online: 'Online',
  Found: 'Found',
  Contact: 'Contact',
  GemShow: 'Gem Show',
  Other: 'Other'
};

// Helper to get description for source type
export const sourceTypeDescriptions: Record<InventorySourceType, string> = {
  Store: 'Physical retail store (rock shop, lapidary supply, etc.)',
  Online: 'Online purchase (eBay, Etsy, website, etc.)',
  Found: 'Collected in the wild (beaches, hiking, mining, etc.)',
  Contact: 'From a person (gift, trade, friend, dealer)',
  GemShow: 'Gem and mineral shows (Tucson, local shows, etc.)',
  Other: 'Other source'
};

File: src/api/MyUglyRocks.Core/Entities/InventorySource.cs
namespace MyUglyRocks.Core.Entities;

/// <summary>
/// Source type for inventory sources (WHERE specimens are acquired from)
/// </summary>
public enum InventorySourceType
{
    Store = 0,      // Physical retail store
    Online = 1,     // Online purchase (eBay, Etsy, etc.)
    Found = 2,      // Collected in the wild (beaches, hiking, etc.)
    Contact = 3,    // From a person (gift, trade, friend, dealer)
    GemShow = 4,    // Gem and mineral shows (Tucson, etc.)
    Other = 5       // Other source
}

/// <summary>
/// Represents WHERE specimens are acquired from (vendors, locations, contacts).
/// This is separate from the actual purchase/acquisition event (Inventory).
/// </summary>
public class InventorySource : BaseEntity
{
    public Guid InventorySourceId { get; set; }
    public Guid UserId { get; set; }
    public InventorySourceType SourceType { get; set; }
    public required string Name { get; set; }
    public string? Location { get; set; }
    public string? Phone { get; set; }
    public string? Url { get; set; }
    public string? ContactName { get; set; }
    public string? Notes { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation properties
    public virtual User User { get; set; } = null!;
    public virtual ICollection<Inventory> Inventories { get; set; } = [];
}
